examples/src/main/java/com/hedera/hashgraph/sdk/examples/MirrorNodeContractQueriesExample.java

```diff
@@ -0,0 +1,193 @@
+
+        /*
+         * Step 1:
+         * Create the contract
+         */
+        System.out.println("Creating new file...");
+        TransactionResponse fileCreateTxResponse = new FileCreateTransaction()
+            // Use the same key as the operator to "own" this file.
+            .setKeys(operatorPublicKey)
+            .setContents(SMART_CONTRACT_BYTECODE)
+            // The default max fee of 1 Hbar is not enough to create a file (starts around ~1.1 Hbar).
+            .setMaxTransactionFee(Hbar.from(2))
+            .execute(client);
+
+        TransactionReceipt fileCreateTxReceipt = fileCreateTxResponse.getReceipt(client);
+        FileId newFileId = fileCreateTxReceipt.fileId;
+
+        Objects.requireNonNull(newFileId);
+        System.out.println("Created new file with ID: " + newFileId);
+
+        var response = new ContractCreateTransaction()
+            .setAdminKey(client.getOperatorPublicKey())
+            .setGas(200000)
+            .setConstructorParameters(new ContractFunctionParameters().addString("Hello from Hedera."))
+            .setBytecodeFileId(newFileId)
+            .setContractMemo("Simple contract with string field")
+            .execute(client);
+
+        var contractId = Objects.requireNonNull(response.getReceipt(client).contractId);
+
+        /*
+         * Step 3:
+         * Wait for mirror node to import data
+         */
+        Thread.sleep(4000);
+
+        /*
+         * Step 4:
+         * Estimate the gas needed
+         */
+        var gas = new MirrorNodeContractEstimateGasQuery()
+            .setContractId(contractId)
+            .setSender(client.getOperatorAccountId())
+            .setGasLimit(30_000)
+            .setGasPrice(1234)
+            .setFunction("getMessage")
+            .execute(client);
+
+        System.out.println("Gas needed for this query: " + gas);
+
+        /*
+         * Step 5:
+         * Do the query against the consensus node using the estimated gas
+         */
+        var callQuery = new ContractCallQuery()
+            .setContractId(contractId)
+            .setGas(gas)
+            .setFunction("getMessage")
+            .setQueryPayment(new Hbar(1));
+
+        var result = callQuery
+            .execute(client);
+
+        /*
+         * Step 6:
+         * Simulate the transaction for free, using the mirror node
+         */
+        var simulationResult = new MirrorNodeContractCallQuery()
+            .setContractId(contractId)
+            .setSender(client.getOperatorAccountId())
+            .setGasLimit(30_000)
+            .setBlockNumber(10000)
+            .setGasPrice(1234)
+            .setFunction("getMessage")
+            .execute(client);
+
+        System.out.println("Simulation result: " + decodeABIHexString(simulationResult));
+        System.out.println("Contract call result: " + result.getString(0));
+    }
+    private static String decodeABIHexString(String hex) {
```

Comment:
You should add a comment letting the users know why they need to decode this result.

Reply:
I added that the mirror node returns the data in ABI Hex format.

Reply:
Yes, that ABI decoding internally would be great.
I think @ochikov and I had chat around this last year.
Maybe something to revisit next year 😆